File: typescript/src/fizzbuzz/fizzbuzz.ts
type Fizzbuzz = number | 'fizz' | 'buzz' | 'fizzbuzz'

function range(start: number, end: number): readonly number[] {
  const size = Math.sqrt((start - end) ** 2) + 1
  const order = start < end ? 1 : -1
  return Array(size)
    .fill(0)
    .map((v, i) => start + i * order)
}

function replace(n: number): Fizzbuzz {
  return n % 15 === 0
    ? 'fizzbuzz'
    : n % 5 === 0
    ? 'buzz'
    : n % 3 === 0
    ? 'fizz'
    : n
}

export function fizzbuzz(start: number, end: number): readonly Fizzbuzz[] {
  return range(start, end).map(replace)
}

File: kotlin/src/main/kotlin/com/goblindegook/fizzbuzz/Fizzbuzz.kt
package com.goblindegook.fizzbuzz

fun range(m: Int, n: Int): IntProgression = if (m < n) m..n else m downTo n

fun fizzbuzz(start: Int, end: Int): List<String> = range(start, end).map {
    when {
      it % 15 == 0 -> "fizzbuzz"
      it % 5 == 0 -> "buzz"
      it % 3 == 0 -> "fizz"
      else -> it.toString()
    }
}
